app-server/app/src/main/scala/scene/Scene.scala

```diff
@@ -0,0 +1,100 @@
+package com.azavea.rf.scene
+
+import com.azavea.rf.utils.Database
+import java.sql.Timestamp
+
+import com.azavea.rf.datamodel.latest.schema.tables._
+import com.azavea.rf.datamodel.latest.enums.JobStatus._
+import com.azavea.rf.datamodel.latest.enums.Visibility._
+
+import scala.concurrent.{Future, ExecutionContext}
+
+import scala.util.{Try, Success, Failure}
+
+/** Handles interaction between scenes and database */
+trait Scene {
+
+  implicit val database:Database
+  implicit val ec:ExecutionContext
+
+  import database.driver.api._
+
+  /** Insert a scene into the database
+    *
+    * @param scene ScenesRow
+    */
+  def insertScene(scene: ScenesRow): Future[Try[ScenesRow]] = {
+    database.db.run {
+      Scenes.forceInsert(scene).asTry
+    } map {
+      case Success(_) => Success(scene)
+      case Failure(e) => Failure(e)
+    }
+  }
+
+  /** Retrieve a single scene from the database
+    *
+    * @param sceneId java.util.UUID ID of scene to query with
+    */
+  def getScene(sceneId: java.util.UUID): Future[Option[ScenesRow]] = {
+    database.db.run {
+      Scenes.filter(_.id === sceneId).result.headOption
+    }
+  }
+
+  /** Delete a scene from the database
+    *
+    * @param sceneId java.util.UUID ID of scene to delete
+    */
+  def deleteScene(sceneId: java.util.UUID): Future[Try[Int]] = {
```

Comment:
It might be cleaner to return `Future[Int]` and throw on failure. Flattening the monad stack out allows cool things in for comprehensions:
```scala
for {
  i <- deleteScene
  () <- sinkFunction(i)
}
```
Flatmapping like this will fail as soon as a non-successful form of the monad is returned (`None`, `Failure`, a `Future` that doesn't return)